cardinal/ecs/persona_integration_test.go

```diff
@@ -0,0 +1,267 @@
+			return true
+		},
+	)
+	assert.NilError(t, err)
+	assert.Equal(t, count, 0) // Assert that no signer components were found
+}
+
+func TestCanAuthorizeAddress(t *testing.T) {
+	// Verify that the CreatePersona is automatically created and registered with a world.
+	world := testutils.NewTestWorld(t).Instance()
+	assert.NilError(t, world.LoadGameState())
+
+	wantTag := "CoolMage"
+	wantSigner := "123_456"
+	ecs.CreatePersonaMsg.AddToQueue(
+		world, ecs.CreatePersona{
+			PersonaTag:    wantTag,
+			SignerAddress: wantSigner,
+		},
+	)
+
+	wantAddr := "0xd5e099c71b797516c10ed0f0d895f429c2781142"
+	ecs.AuthorizePersonaAddressMsg.AddToQueue(
+		world, ecs.AuthorizePersonaAddress{
+			Address: wantAddr,
+		}, &sign.Transaction{PersonaTag: wantTag},
+	)
+	// PersonaTag registration doesn't take place until the relevant system is run during a game tick.
+	assert.NilError(t, world.Tick(context.Background()))
+
+	count := 0
+	q, err := world.NewSearch(ecs.Exact(ecs.SignerComponent{}))
+	assert.NilError(t, err)
+	wCtx := ecs.NewWorldContext(world)
+	err = q.Each(
+		wCtx, func(id entity.ID) bool {
+			count++
+			sc, err := ecs.GetComponent[ecs.SignerComponent](wCtx, id)
+			assert.NilError(t, err)
+			assert.Equal(t, sc.PersonaTag, wantTag)
+			assert.Equal(t, sc.SignerAddress, wantSigner)
+			assert.Equal(t, len(sc.AuthorizedAddresses), 1)
+			assert.Equal(t, sc.AuthorizedAddresses[0], wantAddr)
+			return true
+		},
+	)
+	assert.NilError(t, err)
+	// verify that the query was even ran. if for some reason there were no SignerComponents in the state,
+	// this test would still pass (false positive).
+	assert.Equal(t, count, 1)
+}
+
+func TestCanAuthorizeAddressFailsOnInvalidAddress(t *testing.T) {
+	// Verify that the CreatePersona is automatically created and registered with a world.
+	world := testutils.NewTestWorld(t).Instance()
+	assert.NilError(t, world.LoadGameState())
+
+	wantTag := "CoolMage"
+	wantSigner := "123-456"
+	ecs.CreatePersonaMsg.AddToQueue(
+		world, ecs.CreatePersona{
+			PersonaTag:    wantTag,
+			SignerAddress: wantSigner,
+		},
+	)
+
+	wantAddr := "INVALID ADDRESS"
+	ecs.AuthorizePersonaAddressMsg.AddToQueue(
+		world, ecs.AuthorizePersonaAddress{
+			Address: wantAddr,
+		}, &sign.Transaction{PersonaTag: wantTag},
+	)
+	// PersonaTag registration doesn't take place until the relevant system is run during a game tick.
+	assert.NilError(t, world.Tick(context.Background()))
+
+	count := 0
+	q, err := world.NewSearch(ecs.Exact(ecs.SignerComponent{}))
+	assert.NilError(t, err)
+	wCtx := ecs.NewWorldContext(world)
+	err = q.Each(
+		wCtx, func(id entity.ID) bool {
+			count++
+			sc, err := ecs.GetComponent[ecs.SignerComponent](wCtx, id)
+			assert.NilError(t, err)
+			assert.Equal(t, sc.PersonaTag, wantTag)
+			assert.Equal(t, sc.SignerAddress, wantSigner)
+			assert.Len(t, sc.AuthorizedAddresses, 0) // Assert that no authorized address was added
+			return true
+		},
+	)
+	assert.NilError(t, err)
```

Comment:
im seeing this pattern a lot, and i think it may be useful to make a new test helper: `getSigners() []ecs.SignerComponent { ... }`

that way the `count` variable is just `len` of the slice.